keps/sig-security/2763-ambient-capabilities/README.md

```diff
@@ -0,0 +1,929 @@
+
+#### Story 1
+As a security conscious user I should be able to run my container as non-root and add the minimum set of capabilities required for it to function even when I do not control how the image is built.
+
+#### Story 2
+@Jc2K [commented](https://github.com/kubernetes/enhancements/pull/2757#issuecomment-906729690)
+If you were running one of the popular IDS's or some other network monitoring tool like moloch/arike on your cluster as a `DaemonSet` then i'd expect you could run it as a non root user with CAP_NET_RAW (and possibly CAP_NET_ADMIN).
+
+I personally have run `DaemonSet`'s that would use `CAP_NET_ADMIN` to configure some environment specific networking. I would have run this as non-root if I had this KEP.
+
+Yes, `CAP_NET_ADMIN` is a pretty elevated permission. Most of the use cases I can think are things i'd expect the team deploying the CNI to be interested in, rather than the person deploying something on port 443 (where they are different people). I'd still prefer to deploy them not as root, where that actually works.
+
+#### Story 3
+@Jc2K [commented](https://github.com/kubernetes/enhancements/pull/2757#discussion_r697016426)
+The most useful use of this is where you have to run something that does have to have pretty elevated permissions. Like an IDS, or a network visibility tool, or some daemon that configures some weird aspect of your network stack. You can't get away from the fact that these things are running and they are running with NET_ADMIN or that they are in the host network namespace. Let's set as many restrictions on them as we can and run them as non-root.
+
+### Risks and Mitigations
+One risk of adding this feature is that it could be used to give non-root users elevated permissions on the host. For example one could add `cap_sys_admin` to the ambient capabilities set of a container making it essentially root on host. To mitigate this we propose that for alpha we only support a carefully curated set of "safe" capabilities that are most commonly required by workload.
+
+## Design Details
+
+### Changes to kubernetes API (https://pkg.go.dev/k8s.io/api/core/v1)
+
+<<[UNRESOLVED pick how we want to update the K8S APIs]>>
+
+_Blocking for Alpha._
+
+There are 2 options here:-
+- **Option 1:** Reuse Add field in [Capabilities](https://pkg.go.dev/k8s.io/api/core/v1#Capabilities)
+
+  When a capability gets added explicitly to a non-root container it also gets added to the ambient set in addition to getting added to inheritable, permitted, bounding and effective sets. The default capabilities are not added to the ambient set.
```

Comment:
having capabilities being added to the inheritable set turned out to be a CVE affecting multiple runtimes (Moby, containerd, CRI-O, Podman...), more details in the Moby advisory: https://github.com/moby/moby/security/advisories/GHSA-2mm7-x5h6-5pvq

Having the capabilities added directly to the ambient set requires them to be present in the inheritable set as well, thus getting back the old behavior.

I think this option should not be considered.